app/gui/view/execution-mode-selector/src/play_button.rs

```diff
@@ -0,0 +1,180 @@
+            let width = Var::<Pixels>::from("input_size.x");
+            let time = Var::<f32>::from("input_time");
+            let unit = &width / 16.0;
+            let arc = RoundedArc(&unit * 5.0, (4.0 * FRAC_PI_3).radians(), &unit * 2.0);
+            let rotated_arc = arc.rotate(time * speed);
+            rotated_arc.fill(color).into()
+        }
+    }
+}
+
+
+
+// ===========
+// === FRP ===
+// ===========
+
+ensogl::define_endpoints_2! {
+    Input {
+        reset   (),
+    }
+    Output {
+        pressed (),
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+#[derive(Debug, Clone, CloneRef)]
+pub struct Model {
+    display_object: display::object::Instance,
+    play_icon:      play_icon::View,
+    spinner_icon:   spinner_icon::View,
+}
+
+impl Model {
+    fn update_style(&self, style: &Style) {
+        let triangle_size = Vector2::new(style.triangle_size, style.triangle_size);
+        let padding = Vector2::new(style.padding_x, style.padding_y);
+        let size = triangle_size + 2.0 * padding;
+        self.play_icon.set_size(size);
+        self.spinner_icon.set_size(size);
+        self.play_icon.set_x(-size.x / 2.0 - style.offset);
+        self.spinner_icon.set_x(-size.x / 2.0 - style.offset);
+    }
+
+    fn set_playing(&self, playing: bool) {
+        if playing {
+            self.display_object.remove_child(&self.play_icon);
+            self.display_object.add_child(&self.spinner_icon);
+        } else {
+            self.display_object.remove_child(&self.spinner_icon);
+            self.display_object.add_child(&self.play_icon);
+        }
+    }
+}
+
+impl display::Object for Model {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.display_object
+    }
+}
+
+
+
+// ===================
+// === Play Button ===
+// ===================
+
+impl component::Model for Model {
+    fn label() -> &'static str {
+        "PlayButton"
+    }
+
+    fn new(_app: &Application) -> Self {
+        let display_object = display::object::Instance::new();
+        let play_icon = play_icon::View::new();
+        let spinner_icon = spinner_icon::View::new();
+
+        display_object.add_child(&play_icon);
+
+        Self { display_object, play_icon, spinner_icon }
+    }
+}
+
+impl component::Frp<Model> for Frp {
+    fn init(
+        network: &enso_frp::Network,
+        frp: &<Self as ensogl::application::frp::API>::Private,
+        _app: &Application,
+        model: &Model,
+        style_watch: &StyleWatchFrp,
+    ) {
+        let play_icon = &model.play_icon;
+        let input = &frp.input;
+        let output = &frp.output;
+
+        let style = Style::from_theme(network, style_watch);
+
+        frp::extend! { network
+            eval style.update ((style) model.update_style(style));
+
+            eval_ input.reset (model.set_playing(false));
+
+            output.pressed <+ play_icon.events_deprecated.mouse_down.constant(());
```

Comment:
Instead of using the deprecated events, you may use those handled in the display object hierarchy. See example: https://github.com/enso-org/enso/blob/27b3f75118501bf5e305d9e06b02e7e967b26cc6/lib/rust/ensogl/examples/vector-editor/src/lib.rs#L83 (you may replace `mouse::Up` with `mouse::Down`, of course).